main.cpp

```diff
@@ -568,6 +635,57 @@ void Projects::calculateResearchInvestment(University univName, University::Prof
     std::cout << "This semester were invested on Research $ " << investment << std::endl;
 }
 
+//Free Functions used for 'This' practice
+
+void Market::thisMarket()
+{
+    std::cout << "Market variable call numPeopleWorkingAtStore = " << this->numPeopleWorkingAtStore << std::endl;
+
+    std::cout << "Market function call adjustInventary() = " << this->adjustInventary(phillip) << std::endl;
+}
+
+void Market::Customer::thisMarketCustomer()
+{
+    std::cout << "Market::Customer variable call productsToOrder = " << this->productsToOrder << std::endl;
+    
+    std::cout << "Market::Customer function call countNumberCustomers() = " << this->countNumberCustomers(3) << std::endl;
+}
+
+void University::thisUniversity()
+{
+    std::cout << "University variable call semIncome = " << this->numClassrooms << std::endl;
+
+    std::cout << "University function call teachStudents() = " << this->teachStudents() << std::endl;
+}
+
+void University::Professor::thisUniversityProfessor()
+{
+    std::cout << "University::Professor variable call professorName  = " << this->professorName << std::endl;
+
+    std::cout << "University::Professor function call checkSubscribedStudents()  = " << this->checkSubscribedStudents() << std::endl;
+}
+
+void University::Student::thisUniversityStudent()
+{
+    std::cout << "University::Student variable call weeklyResearchStudyHours  = " << this->weeklyResearchStudyHours << std::endl;
+
+    std::cout << "University::Student function call computeWeekStudyTime()  = " << this->computeWeekStudyTime() << std::endl;
+}
+
+void Computer::thisComputer()
+{
+    std::cout << "Computer variable call execTask  = " << this->execTask << std::endl;
+
+    std::cout << "Computer function call saveInfo()  = " << this->saveInfo() << std::endl;
+}
+
+void Computer::Hardware::thisComputerHardware()
+{
+    std::cout << "Computer::Hardware variable call screen = " << this->screen << std::endl;
+    
+    std::cout << "Computer::Hardware function call playGames() = " << this->playGames() << std::endl;
+}
```

Comment:
None of these function names conform to the mantra of 'Name Functions what they do'.
None of these function names contain a verb. 
- [ ] revise